### ISSUE TITLE
Support of [TOC] to show table of contents

### ISSUE THREAD

**Issue by MelBourbon** [GitJournal/GitJournal#88]
Support of `[TOC]` or similar to show a table of contents would really great and is broadly used and appreciated by using markdown notes.

**Comment by vHanda** [comment]
Hi. Do you mean auto generated or manual? If it's the latter, that would mean the ability to link to other markdown files, right?

**Comment by MelBourbon** [comment]
No, for the start it would be fine to use it within one notes. By adding [TOC] at the top of the note the table of contents is rendered based on the headlines in the note.

Below example from Markor

![Screenshot_20200326-092543__01](https://user-images.githubusercontent.com/30600002/77625450-e3966e00-6f43-11ea-89a9-38bfde39fbaa.jpg)
![Screenshot_20200326-092548__01](https://user-images.githubusercontent.com/30600002/77625463-e85b2200-6f43-11ea-863a-f3e04825b80a.jpg)

**Comment by MelBourbon** [comment]
> If it's the latter, that would mean the ability to link to other markdown files, right?

But yes, this would be great also and maybe fulfill another use case where user are creating more small seperate notes. But I think this is technically more difficult to realise.

**Comment by vHanda** [comment]
Ah. I understand now.

Generating it seems quite simple, though making it link to those relevant headers seems more challenging. Let me see what I can do.

For my own reference: https://github.com/shd101wyy/markdown-preview-enhanced/blob/master/docs/toc.md

**Comment by vHanda** [comment]
Note to self: Use https://www.markdownguide.org/extended-syntax/#heading-ids
